### PR TITLE
FEATURE: RAIL-2103 Add new interfaces for Analytical Dashboard objects

### PR DESCRIPTION
**Pull request by Tulean**

### CODE REVIEW

**Comment by yenkins** [comment]
Currently destined for [release R199](https://confluence.intgdc.com/display/Dev/R199).
This release is **zero downtime** release.
Code drop: 2020-02-27 00:00.

**Comment by yenkins** [comment]
Story [RAIL-2103](http://jira.intgdc.com/browse/RAIL-2103) Add new interfaces for Analytical Dashboard Objects
	Status: New
	Priority: Major
	Assignee: Tuan Anh Le

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/49/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/127/ : SUCCESS in 58s

File: src/KPI.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ];
+    drillTo?: IKpiProjectDashboardLink;
+    dateDimension?: string;
+    dateDataSet?: string;
+}
+
+export interface IKpiContentWithoutComparison {
+    metric: string;
+    comparisonType: IKpiComparisonTypeNoComparison;
+    ignoreDashboardFilters: [
```
Comment (no23reason) [inline src/KPI.ts]: Should probably be
```ts
Array<ExtendedDateFilters.IDateFilterReference | ExtendedDateFilters.IAttributeFilterReference>
```

File: src/visualizationWidget.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import { IObjectMeta } from './Meta';
+import { ExtendedDateFilters } from './ExtendedDateFilters';
+import { VisualizationObject } from './VisualizationObject';
+export interface IVisualizationWidget {
+    meta: IObjectMeta;
+    content: {
+        visualization: string;
+        dateDataSe?: string;
+        ignoreDashboardFilters: [
```
Comment (no23reason) [inline src/visualizationWidget.ts]: Array type is wrong, see my other comment, please.

File: src/KPI.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export interface IKPI {
+    meta: IObjectMeta;
+    content: IKpiContentWithoutComparison | IKpiContentWithComparison;
+}
+
+export interface IKpiContentWithComparison {
+    metric: string;
+    comparisonType: IKpiComparisonTypeComparison;
+    comparisonDirection: IKpiComparisonDirection;
+    ignoreDashboardFilters: [
```
Comment (no23reason) [inline src/KPI.ts]: Array type is wrong, see my other comment, please.

File: src/KPI.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import { IObjectMeta } from './Meta';
+import { ExtendedDateFilters } from './ExtendedDateFilters';
+
+export interface IKPI {
+    meta: IObjectMeta;
+    content: IKpiContentWithoutComparison | IKpiContentWithComparison;
+}
+
+export interface IKpiContentWithComparison {
```
Comment (no23reason) [inline src/KPI.ts]: Suggestion: you could probably extract properties that are in both `IKpiContentWithComparison` and `IKpiContentWithoutComparison` into something like `IKpiContentBase` to reduce duplication. You can then write 

```ts
export interface IKpiContentWithComparison extends IKpiContentBase {
  comparisonType: IKpiComparisonTypeComparison;
  comparisonDirection: IKpiComparisonDirection;
}
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/50/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/128/ : SUCCESS in 46s

**Comment by yenkins** [comment]
Story [RAIL-2103](http://jira.intgdc.com/browse/RAIL-2103) Add new interfaces for Analytical Dashboard Objects
	Status: Hacking
	Priority: Major
	Assignee: Tuan Anh Le

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/51/ : UNSTABLE in 17s
- gooddata-typings-unit-tests-zuul-docker-el7 : SKIPPED

**Comment by no23reason** [comment]
ok to test

**Comment by no23reason** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/54/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/131/ : SUCCESS in 1m 04s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-gate/22/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/132/ : SUCCESS in 28s

**Pull request merged**